### PR TITLE
docs: add ecrecover example

### PR DESCRIPTION
**Pull request by sarahschwartz**
adds an example for using `ecrecover`

### CODE REVIEW

File: content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md [hunk 1]
```diff
@@ -106,3 +158,22 @@ bytes memory input = abi.encodePacked(
...
+const isValidMessageSignature = await utils.isMessageSignatureCorrect(provider, ADDRESS, message, messageSignature);
+
+const isValidTypesSignature = await utils.isTypedDataSignatureCorrect(provider, ADDRESS, await eip712Signer.getDomain(), utils.EIP712_TYPES, EIP712Signer.getSignInput(tx), typedSignature);
+```
+
+Currently these methods only support verifying ECDSA signatures, but very soon they will support EIP1271 signature verification as well.
+
+Both of these methods return `true` or `false` depending on whether the message signature is correct.
+
+It is **not recommended** to use the `ethers.js` library to verify user signatures.
```
Comment (uF4No) [inline content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md]: why?

File: content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md [hunk 2]
```diff
@@ -106,3 +158,22 @@ bytes memory input = abi.encodePacked(
...
+
+```ts
+import { utils, EIP712Signer } from "zksync-ethers";
+
+const isValidMessageSignature = await utils.isMessageSignatureCorrect(provider, ADDRESS, message, messageSignature);
+
+const isValidTypesSignature = await utils.isTypedDataSignatureCorrect(provider, ADDRESS, await eip712Signer.getDomain(), utils.EIP712_TYPES, EIP712Signer.getSignInput(tx), typedSignature);
+```
+
+Currently these methods only support verifying ECDSA signatures, but very soon they will support EIP1271 signature verification as well.
```
Comment (uF4No) [inline content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md]: can you confirm this is still true?

Comment (sarahschwartz) [reply content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md]: good catch, it is supported now

File: content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md (hunk 1 above)
Comment (sarahschwartz) [reply content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md]: good question, I updated this to explain why

File: content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md [hunk 3]
```diff
@@ -55,24 +55,76 @@ contract TestSignatureChecker {
 }
 ```
 
-## Verifying AA signatures
+## Validating Signatures with ECRecover
 
-Our SDK provides two methods within `utils` to verify the signature of an account:
-[`isMessageSignatureCorrect`](https://sdk.zksync.io/js/ethers/api/v6/utilities#ismessagesignaturecorrect) and [`isTypedDataSignatureCorrect`](https://sdk.zksync.io/js/ethers/api/v6/utilities#istypeddatasignaturecorrect).
+The [`ecrecover`](https://docs.zksync.io/zk-stack/components/prover/circuits/ecrecover)
```
Comment (MexicanAce) [inline content/00.build/65.developer-reference/40.account-abstraction/50.signature-validation.md]: Should this be a relative path?
```suggestion
The [`ecrecover`](/zk-stack/components/prover/circuits/ecrecover)
```

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**